libs/books/data-access/src/lib/+state/reading-list.effects.ts

```diff
@@ -54,9 +66,74 @@ export class ReadingListEffects implements OnInitEffects {
     )
   );
 
+  undoAddtoReadingList$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(ReadingListActions.confirmedAddToReadingList),
+      filter(({book}) => book.isOpenSnackBar),
+      switchMap(({ book }) => {
```

Comment:
switchMap is mostly used for typeahead  kind of scenario where we want to cancel previously triggered request and want to start new one. Try different operator instead of this.